core/src/main/java/com/graphhopper/util/PathSimplification.java

```diff
@@ -99,16 +99,13 @@ public PointList simplify() {
                     if (removed > 0) {
                         for (int i = 0; i < listsToSimplify.size(); i++) {
                             reduceLength(listsToSimplify.get(i), offset[i], startIntervals[i], endIntervals[i] - removed);
-                            /*
-                            // TODO This is not needed, I left it here temporarily, has to be removed before merging
-                            if(listsToSimplify.get(i).get(0) instanceof PathDetail){
+                            if (listsToSimplify.get(i).get(0) instanceof PathDetail) {
```

Comment:
Ah, because for the remaining PathDetails we need to update the indices and instructions have the pointlist stored and do not need an update if the length changes